### ISSUE TITLE
Tracking Server start/stop/restart/pause/list

### ISSUE THREAD

**Issue by khikmatullaev** [mlflow/mlflow#1856]
During the exploitation of MLFLow I faced the problem with starting/stoping MLFlow Tracking Server. I did not find any information in the documentation about no rough stopping(pkill command) of Tracking Server. From my point of view, will be great if the following functionality will be added in the next versions.

Server Stop 
```
mlflow server stop
```

Server Restart
```
mlflow server restart
```

Server Restart with --time parameter in seconds or milliseconds
```
mlflow server pause --time 100
```

And something like

```
mlflow server list
```
Server List that return all information about stared mlflow server.

**Comment by dingobar** [comment]
I would very much like this!

**Comment by calebmarchent** [comment]
Agreed, it would be a lot cleaner to have an "official" stop process; ensuring that in-process jobs complete etc; as at the moment am stuck using kill; which itself seems to leave the gnuicorn processes going.

**Comment by smurching** [comment]
@khikmatullaev thanks for filing this (& thanks all for chiming in)! It'd be good to write out some of the use cases here before proceeding, would love your help with that - from an initial glance, it seems some things this proposal would enable are:
* More easily writing a script for upgrading the MLflow server (stop existing server, upgrade MLflow version, start new server)
* Under the hood, ensuring requests are properly drained etc before stopping the server

Is ^that accurate? & are there additional use cases this feature would enable?

**Comment by smurching** [comment]
@SkyWaffles thanks for the input! CTRL+C should kill the gunicorn processes - if not, that might be a separate bug/issue.

As next steps for this feature, we should:
* Write out use cases for this feature (can be brief). So far, the main use cases I'm aware of are finding/referencing `mlflow server` processes and stopping said processes (not sure what use case for the `--pause` functionality is)
* Write a design proposal outlining the new `mlflow server` commands and their behavior/arguments in a bit more detail. All new commands should be motivated by use cases from the previous step
* Actually open, review, & merge a PR

Would be happy to review/provide feedback during each of the above steps :)

**Comment by smurching** [comment]
Applying the important-longterm issue for now, can bump the priority if someone's interested in writing up use cases/a detailed design for this feature

**Comment by mikolajkania** [comment]
Any news on it? It is weird this option is not available in such mature project.

**Comment by chenmoneygithub** [comment]
I am not familiar with our strategy on OSS MLflow server, @harupy @dbczumar is this something we want to add? Considering its popularity, I am not going to close this issue despite it's pretty old.
